lib/sqlalchemy/dialects/sqlite/base.py

```diff
@@ -402,6 +402,253 @@ def set_sqlite_pragma(dbapi_connection, connection_record):
         PRIMARY KEY (id) ON CONFLICT FAIL
     )
 
+.. _sqlite_insert_on_conflict
+
+INSERT...ON CONFLICT (Upsert)
+-----------------------------------
+From version 3.24.0 onwards, SQLite supports "upserts" (update or insert)
+of rows into a table via the ``ON CONFLICT`` clause of the ``INSERT``
+statement. A candidate row will only be inserted if that row does not violate
+any unique constraints. In the case of a unique constraint violation, a
```

Comment:
**mike bayer** (zzzeek) wrote:

"unique or primary key constraints".

perhaps we can use the verbiage from the PostgreSQL dialect here?  the PG form of this construct doesn't seem to be different.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2230